modules/post/windows/gather/enum_services.rb

```diff
@@ -56,71 +61,74 @@ def run
-        'Header'     => 'Services',
-        'Indent'     => 1,
-        'SortIndex'  => 0,
-        'Columns'    => ['Name', 'Credentials', 'Command', 'Startup']
+      'Header' => 'Services',
+      'Indent' => 1,
+      'SortIndex' => 0,
+      'Columns' => ['Name', 'Credentials', 'Command', 'Startup']
     )
 
-    print_status("Listing Service Info for matching services, please wait...")
-    service_list.each do |srv|
+    print_status('Listing Service Info for matching services, please wait...')
+
+    services = service_list
+
+    vprint_status("Found #{services.length} Windows services")
+
+    services.each do |srv|
       srv_conf = {}
 
       # make sure we got a service name
-      if srv[:name]
-        begin
-          srv_conf = service_info(srv[:name])
-          if srv_conf[:startname]
-            # filter service based on filters passed, the are cumulative
-            if qcred && !srv_conf[:startname].downcase.include?(qcred)
-              next
-            end
-
-            if qpath && !srv_conf[:path].downcase.include?(qpath)
-              next
-            end
-
-            # There may not be a 'Startup', need to check nil
-            if qtype && !(START_TYPE[srv_conf[:starttype]] || '').downcase.include?(qtype)
-              next
-            end
-
-            # count the occurance of specific credentials services are running as
-            serviceCred = srv_conf[:startname].upcase
-            unless serviceCred.empty?
-              if credentialCount.has_key?(serviceCred)
-                credentialCount[serviceCred] += 1
-              else
-                credentialCount[serviceCred] = 1
-                # let the user know a new service account has been detected for possible lateral
-                # movement opportunities
-                print_good("New service credential detected: #{srv[:name]} is running as '#{srv_conf[:startname]}'")
-              end
-            end
-
-            results_table << [srv[:name],
-                              srv_conf[:startname],
-                              START_TYPE[srv_conf[:starttype]],
-                              srv_conf[:path]]
-          end
+      unless srv[:name]
+        print_error("Problem retrieving service information - no name found for service: #{srv}")
+        next
+      end
+
+      begin
+        srv_conf = service_info(srv[:name])
+
+        next unless srv_conf[:startname]
+
+        # filter service based on provided filters
+        next if qcred && !srv_conf[:startname].downcase.include?(qcred)
+        next if qpath && !srv_conf[:path].downcase.include?(qpath)
+
+        # There may not be a 'Startup', need to check nil
+        start_type = srv_conf[:starttype]
+        start_type = start_type.blank? ? nil : START_TYPE[start_type]
 
-        rescue RuntimeError => e
-          print_error("An error occurred enumerating service: #{srv[:name]}: #{e}")
+        next if qtype && !start_type.downcase.include?(qtype)
```

Comment:
Several things wrong here:

1. Why are you checking to see if the value is blank only to set it to `nil` when  calling `.downcase` on `nil` will result in a stack trace since `nil` doesn't have a `.downcase` method?
2. What happens if `start_type` doesn't contain a value within the `START_TYPE` array? This will just return `nil` as well, which isn't being accounted for here.

Reply:
Fixed by https://github.com/rapid7/metasploit-framework/pull/16929/commits/6cafcb156d45b9d739046d2534f44c8eaecca93d